### PR TITLE
Add missing template name and line no to some deprecation messages

### PR DESCRIPTION
**Pull request by yceruto**
before:
twig-before

after:
twig-after

### CODE REVIEW

**Comment by stof** [comment]
Wasn't it done on purpose in the past to avoid breaking the grouping @nicolas-grekas ?

**Comment by nicolas-grekas** [comment]
Yes, that's true. The goal is to limit the number of repetitive messages.
But I guess there is not a huge list of different file+line, so this should be fine?

**Comment by stof** [comment]
Well, every time this deprecation is triggered, the file+line will be different (at least if you write your templates in a readable way rather than putting them in a single very long line).

**Comment by yceruto** [comment]
I'm fine with adding the template name only if we want to keep the group per template

**Comment by yceruto** [comment]
Note that the real problem here is that neither "Show context" nor "Show trace" show the affected template for those deprecation notices, so it's very hard to find which template contains the error.

**Comment by fabpot** [comment]
Not related, but I've just released a new version of SwiftmailerBundle to get rid of these deprecation notices.

**Comment by fabpot** [comment]
I think we need to add this information as Twig templates are complied to PHP, so without doing it, it's quite complex to find the usage (at least if you don't know how to use grep :)). So 👍 on my side.

**Review by nicolas-grekas — APPROVED** [review]

File: src/TokenParser/FilterTokenParser.php [hunk 1]
```diff
@@ -30,21 +30,24 @@ final class FilterTokenParser extends AbstractTokenParser
 {
     public function parse(Token $token)
     {
-        @trigger_error('The "filter" tag is deprecated since Twig 2.9, use the "apply" tag instead.', E_USER_DEPRECATED);
+        $stream = $this->parser->getStream();
+        $lineno = $token->getLine();
+
+        @trigger_error(sprintf('The "filter" tag is deprecated since Twig 2.9, use the "apply" tag instead. ("%s" at line %d).', $stream->getSourceContext()->getName(), $lineno), E_USER_DEPRECATED);
```
Comment (fabpot) [inline src/TokenParser/FilterTokenParser.php]: I would use `tag instead in "%s" at line %d.` to be more similar to what we're doing with exceptions.

Comment (fabpot) [reply src/TokenParser/FilterTokenParser.php]: Or perhaps like you did elsewhere here: `The "filter" tag in "%s" at line %d is deprecated ...`

Comment (yceruto) [reply src/TokenParser/FilterTokenParser.php]: done.

**Comment by fabpot** [comment]
Thank you @yceruto.

**Pull request merged**